src/bin/main.rs

```diff
@@ -298,78 +327,74 @@ fn main() {
+    if let Some(code_dir) = code_dir {
+        if let Some(code_file_patterns) = &paths.code_files {
+            copy_files(code_file_patterns, &paths.code_paths, code_dir)?;
+        }
+    }
+
+    if let Some(doc_dir) = doc_dir {
+        if let Some(doc_file_patterns) = &paths.doc_files {
+            copy_files(doc_file_patterns, &paths.doc_paths, doc_dir)?;
+        }
+    }
+
+    return Ok(());
+}
+
+fn copy_files(
+    patterns: &[String],
+    path_mod: &Option<Vec<String>>,
+    target_dir: PathBuf,
+) -> Result<(), String> {
+    match path_mod {
+        Some(path_mod) if patterns.len() != path_mod.len() => {
+            return Err(
+                "If argument code_paths/doc_paths is given in the toml file, it must have as many elements as argument code_files/doc_files".to_string()
             );
-            return;
         }
         _ => (),
     }
-
     let mut track_copy_dest: HashMap<PathBuf, PathBuf> = HashMap::new();
-    if let Some(code_dir) = code_dir {
-        if let Some(code_file_patterns) = &paths.code_files {
-            for (idx, code_file_pattern) in code_file_patterns.iter().enumerate() {
-                let code_path = paths.code_paths.as_ref().map(|code_paths| &code_paths[idx]);
-                for code_file in PathUtil::list_files(code_file_pattern).unwrap() {
-                    let out_path = code_path.map_or(code_file.clone(), |code_path| {
-                        modify_path(&code_file, &code_path)
-                    });
-                    match track_copy_dest.entry(out_path.clone()) {
-                        Occupied(entry) => {
-                            eprintln!(
-                                "ERROR: Attempted to copy multiple code files to {}: from {} and {}",
-                                out_path.display(), entry.get().display(), code_file.display()
-                            );
-                            return;
-                        }
-                        Vacant(entry) => {
-                            entry.insert(code_file.clone());
-                        }
-                    }
-                    eprintln!(
-                        "Copying code file {} to {}",
-                        code_file.display(),
-                        out_path.display()
-                    );
-
-                    let mut file_path = code_dir.clone();
-                    file_path.push(out_path);
-
-                    fs::create_dir_all(file_path.parent().unwrap()).unwrap();
-                    match fs::copy(&code_file, &file_path) {
-                        Ok(_) => {}
-                        Err(err) => eprintln!(
-                            "ERROR: --> Error copying code file {}: {}",
-                            code_file.display(),
-                            err
-                        ),
-                    }
+    for (idx, file_pattern) in patterns.iter().enumerate() {
+        let code_path = path_mod.as_ref().map(|code_paths| &code_paths[idx]);
```

Comment:
I think it would improve readability to excise all `code_*` prefix from this now general code?